### PR TITLE
[Improvement][LocalOrder] Add tests about keeping consistent with FixedSize when no skew optimization

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
1. Introduce the data length check in LocalOrder strategy
2. Add tests about keeping consistent with FixedSize Strategy when no skew optimization

### Why are the changes needed?
More stability

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UTs

### CODE REVIEW

**Comment by zuston** [comment]
PTAL @jerqi

File: common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java [hunk 1]
```diff
@@ -105,7 +107,11 @@ public List<ShuffleDataSegment> split(ShuffleIndexResult shuffleIndexResult) {
 
         // If ShuffleServer is flushing the file at this time, the length in the index file record may be greater
         // than the length in the actual data file, and it needs to be returned at this time to avoid EOFException
-        if (dataFileLen != -1 && totalLen >= dataFileLen) {
+        if (dataFileLen != -1 && totalLen > dataFileLen) {
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java:110]: Do we remove the `=`?

Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java:110]: Yes. we need. This has been discussed in #320

File: common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java [hunk 2]
```diff
@@ -105,7 +107,11 @@ public List<ShuffleDataSegment> split(ShuffleIndexResult shuffleIndexResult) {
...
+          long mask = (1L << Constants.PARTITION_ID_MAX_LENGTH) - 1;
+          LOGGER.warn("Abort inconsistent data, the data length: {}(bytes) recorded in index file is greater than "
+                  + "the real data file length: {}(bytes). Partition id: {}. This should not happen.",
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java:113]: I feel that it will happen when the data is flushing.

Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java:113]: Actually I never encounter this problem with localfile storage

Comment (jerqi) [reply common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java:113]: > Actually I never encounter this problem with localfile storage

How about HDFS?

Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java:113]: Maybe... Overall this is not a breaking fix.

Comment (jerqi) [reply common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java:113]: Just modify the log information.

Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java]: Done

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston

**Comment by jerqi** [comment]
There are conflicts between LOCAL_ORDER and getInMemoryShuffleData. `getInMemoryShuffleData` need that we don't change the order. Because we use lastBlockId to read data.

**Comment by zuston** [comment]
> There are conflicts between LOCAL_ORDER and getInMemoryShuffleData. `getInMemoryShuffleData` need that we don't change the order. Because we use lastBlockId to read data.

You are right. I will fix it by creating a reference copy to avoid breaking down original sequence.

**Comment by zuston** [comment]
> > There are conflicts between LOCAL_ORDER and getInMemoryShuffleData. `getInMemoryShuffleData` need that we don't change the order. Because we use lastBlockId to read data.
> 
> You are right. I will fix it by creating a reference copy to avoid breaking down original sequence.

Fixed https://github.com/apache/incubator-uniffle/pull/336

**Pull request merged**